src/main/java/ml/compute/InputData.java

```diff
@@ -0,0 +1,28 @@
+package ml.compute;
+
+import java.util.Set;
+
+import ml.preprocessor.types.PreprocessedData;
+import ml.preprocessor.types.RawData;
+
+/**
+ * Created by mikeecb on 17/10/2015.
+ */
+public class InputData {
```

Comment:
That is what I had in mind. And I think Input Data should have an accept. So we also need a generic inputDataVisitor. Or do you want to move away from java and do something fancy in scala? But I don't know anything about that.